silkrpc/core/gas_price_oracle_test.cpp

```diff
@@ -0,0 +1,593 @@
+/*
+    Copyright 2021 The Silkrpc Authors
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#include "gas_price_oracle.hpp"
+#include <boost/endian/conversion.hpp>
+
+#include <algorithm>
+#include <iostream>
+#include <catch2/catch.hpp>
+#include <asio/co_spawn.hpp>
+#include <asio/thread_pool.hpp>
+#include <asio/use_future.hpp>
+
+#include <evmc/evmc.hpp>
+
+#include <silkrpc/types/block.hpp>
+
+namespace silkrpc {
+
+using evmc::literals::operator""_address;
+
+static std::vector<silkworm::BlockWithHash> kBlocks;
+
+static const evmc::address kBeneficiary = 0xe5ef458d37212a06e3f59d40c454e76150ae7c31_address;
+static const evmc::address kFromTnx1 = 0xe5ef458d37212a06e3f59d40c454e76150ae7c32_address;
+static const evmc::address kFromTnx2 = 0xe5ef458d37212a06e3f59d40c454e76150ae7c33_address;
+
+static silkworm::BlockWithHash allocate_block(uint64_t block_number,
+    const evmc::address &beneficiary,
+    const intx::uint256 &base_fee,
+    const intx::uint256 &max_priority_fee_per_gas_tx1,
+    const intx::uint256 &max_fee_per_gas_tx1,
+    const intx::uint256 &max_priority_fee_per_gas_tx2,
+    const intx::uint256 &max_fee_per_gas_tx2) {
+
+    silkworm::BlockWithHash block_with_hash;
+
+    block_with_hash.block.header.number = block_number;
+    block_with_hash.block.header.beneficiary = beneficiary;
+    block_with_hash.block.header.base_fee_per_gas = base_fee;
+
+    block_with_hash.block.transactions.resize(2);
+    block_with_hash.block.transactions[0].max_priority_fee_per_gas = max_priority_fee_per_gas_tx1;
+    block_with_hash.block.transactions[0].max_fee_per_gas = max_fee_per_gas_tx1;
+    block_with_hash.block.transactions[0].from = kFromTnx1;
+
+    block_with_hash.block.transactions[1].max_priority_fee_per_gas = max_priority_fee_per_gas_tx2;
+    block_with_hash.block.transactions[1].max_fee_per_gas = max_fee_per_gas_tx2;
+    block_with_hash.block.transactions[1].from = kFromTnx2;
+
+    return block_with_hash;
+}
+
+static void clear_blocks_vector() {
+    kBlocks.clear();
+}
+
+static void fill_blocks_vector(uint16_t size,
```

Comment:
Add std::vector<<silkworm::BlockWithHash>> as first paameter and do not use `kBlocks`